### ISSUE TITLE
Cannot chain events

### ISSUE THREAD

**Issue by Arrow7000** [RIAEvangelist/event-pubsub#5]
Usually with objects that emit events you can string them together like this:

```
stream
    .on('error', err => {
        console.log('stream error:', err);
        thing.trigger('error', err);
    })
    .on('end', () => {
        console.log('Stream end: ' + fileName + ' successfully sent to server');
        thing.trigger('end', fileName);
    });
```
I tried to do this with event-pubsub like this:

```
let thing = {};
new Pubsub(thing);
let stream = ss.createStream();
ss(socket).emit('crash-video', stream, { name: fileName });
let file = fs.createReadStream(fileName);
stream
    .on('error', err => {
        console.log('stream error:', err);
        thing.trigger('error', err);
    })
    .on('end', () => {
        console.log('Stream end: ' + fileName + ' successfully sent to server');
        thing.trigger('end', fileName);
    });

thing
    .on('error', err => {
        console.log('Error:', err);
    })
    .on('end', fileName => {
        console.log('Stream end: ' + fileName + ' successfully sent to server');
    });
```

But that didn't work. It throws the following error:
```
C:\Users\aadler\Documents\Programming projects\Tracker-3\device\index.js:183
        .on('end', fileName => {
        ^

TypeError: Cannot read property 'on' of undefined
    at streamFile (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\index.js:183:9)
    at Socket.<anonymous> (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\index.js:104:13)
    at Socket.Emitter.emit (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\socket.io-client\node_modules\component-emitter\index.js:131:20)
    at Socket.emit (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\socket.io-client\lib\socket.js:133:10)
    at Socket.onconnect (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\socket.io-client\lib\socket.js:321:8)
    at Socket.onpacket (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\socket.io-client\lib\socket.js:217:12)
    at Manager.<anonymous> (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\component-bind\index.js:21:15)
    at Manager.Emitter.emit (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\socket.io-client\node_modules\component-emitter\index.js:131:20)
    at Manager.ondecoded (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\socket.io-client\lib\manager.js:333:8)
    at Decoder.<anonymous> (C:\Users\aadler\Documents\Programming projects\Tracker-3\device\node_modules\component-bind\index.js:21:15)
```
It only works if I attach an event handler to the pubsubbed object directly repeatedly, like this:
```
    thing.on('error', err => {
        console.log('Error:', err);
    })

    thing.on('end', fileName => {
        console.log('Stream end: ' + fileName + ' successfully sent to server');
    });
```
Am I doing something wrong or does this library not support this kind of chaining?

**Comment by RIAEvangelist** [comment]
Nice request! We can do this.

**Comment by RIAEvangelist** [comment]
available in v3.0.0
